### ISSUE TITLE
Slow when used with a lot of pages

### ISSUE THREAD

**Issue by maartenhunink** [kylephillips/wp-nested-pages#211]
I'm currently working on a website which has a lot of pages ( > 3000, I know, crazy). The plugin becomes unresponsive and very slow.

**Comment by amityweb** [comment]
I'm going to add my voice to this too... we've used Nested Pages for many years, but I've got a new site with over 1600 pages, and those pages are children of approx. 20 parent pages. Nested Pages is no longer useable, its way to slow. I advise the subpages are only loaded on AJAX when expanded. Reordering needs to happen quicker. I am assuming it must be processing every single page (parent and child) when reordering which is why it takes so long. Needs addressing. In the meantime we use Swifty Page or CMS Tree View, but I prefer Nested Pages.

**Comment by maartenhunink** [comment]
Yes, same here. CMS Tree View works because it loads the pages with ajax,
but Nested Pages would be so much better.

**Comment by fubizitch** [comment]
Agreed.  Needs to load on the fly rather than all at once.  We have 800 pages, and it takes 12 seconds to load the Pages view each time.  Any way to cache this???

**Comment by fubizitch** [comment]
By the way, our upgrade from PHP 5.4 (with opcache) to PHP 7.2 made this perform over 3x faster!  Huge difference.  Perhaps with some extra PHP optimization, this can perform even faster?

**Comment by GiantKiwi** [comment]
Nearly 500 pages on our site, and query monitor is telling me the query is taking 5.9 seconds per page on PHP7.2

**Comment by lkraav** [comment]
I'm fairly certain one just can't load that much information in one request.

Lazy loading via REST API, smart caching (hard problem), or something of that sort maybe has to get involved.

**Comment by workplaceninja** [comment]
No update on this?

**Comment by lkraav** [comment]
I could likely talk my shareholders into crowdfunding a campaign to add paging to this plugin.

OTOH, many plugins in WP ecosystem are already or will be refactoring their admin UI towards React. It seems like https://github.com/WP-API/new-list-tables is archived, I wonder what the most up to date public effort towards this might be?

**Comment by PLUS-RobertEhrenleitner** [comment]
Hello, we have a solution to this problem implemented on our install. It consists of providing a dropdown box as soon as more than 3 pagegroups (i.e. toplevel pages) exist. I will provide an pull request as soon as I have time.
We also provided a solution to a problem that arises due to a restriction of many webservers when reordering a page on a website with a lots of pages.

**Comment by koch-crsd** [comment]
@PLUS-RobertEhrenleitner what's your mentioned solution for this? Can you please share? :) Much appreciated!

**Comment by PLUS-RobertEhrenleitner** [comment]
Hello, I am sorry, we are completely overloaded. I would much appreciate contributing, since we have to do the same hassle on each update. Hopefully, I can come up with a contribution until March 2023.

**Comment by PLUS-RobertEhrenleitner** [comment]
I finally proposed a solution with [PR 335](https://github.com/kylephillips/wp-nested-pages/pull/335).
